peps/pep-0781.rst

```diff
@@ -0,0 +1,280 @@
+existing zlib-based compression. While this PEP does *not* prescribe any
+changes to the wheel format or other packaging standards, having Zstandard
+bindings in the standard library would enable a future PEP to improve the user
+experience for Python wheel packages.
+
+Rationale
+=========
+
+Implementation based on ``pyzstd``
+----------------------------------
+
+The implementation for this PEP is based on the `pyzstd project <https://github.com/Rogdham/pyzstd>`_.
+This project was chosen as the code was `originally written to be upstreamed <https://github.com/python/cpython/issues/81276#issuecomment-1093824963>`_
+to CPython by Ma Lin, who also wrote the `output buffer implementation used in
+the standard library today <https://github.com/python/cpython/commit/f9bedb630e8a0b7d94e1c7e609b20dfaa2b22231>`_.
+The project has since been taken over by Rogdham and is published to PyPI. The
+APIs in ``pyzstd`` are similar to the APIs for other compression modules in the
+standard library such as ``bz2`` and ``lzma``.
+
+Changes and removals to ``pyzstd``'s APIs
+-----------------------------------------
+
+n.b. maybe this should be an appendix?
+
+To keep the initial implementation simple and make it easier to review, several
+APIs were modified or removed completely. The "RichMem" API is removed as
+CPython's output buffer does not use `mremap(2) <https://www.man7.org/linux/man-pages/man2/mremap.2.html>`_.
+This could be integrated into CPython in a future change benefitting all
+compression libraries, but is not necessary for the initial introduction of
+Zstandard. The ``ZstdFile`` implementation was re-written in Python to match
+``lzma`` and other compression modules, and reduce the amount of C code in need
+of review.
+
+The ``compress_stream`` / ``decompress_stream`` functions were removed, as they
+were performance optimizations and can be replaced with using the ``open``
+function from the Zstandard module.
+
+The other major change is the ``level_or_options`` argument was split into two
+independent arguments to keep the argument parsing clearer and improve clarity
+of usage.
+
+Finally, features requiring newer versions of Zstandard were removed, which
+is mostly the support for the ``ZSTD_c_targetCBlockSize`` compression
+parameter.
+
+Minimum supported Zstandard version
+-----------------------------------
+
+The minimum supported Zstandard was chosen as v1.4.5, released in May of 2020.
+This version was chosen as a minimum based on reviewing the versions of
+Zstandard available in a number of Linux distribution package repositories,
+including LTS releases. This version choice is rather conservative to maximize
+compatibility with existing LTS Linux distributions, but a newer Zstandard
+version could likely be chosen given that newer Python releases are generally
+packaged as part of newer distribution releases.
+
+Introduction of a ``compression`` namespace
+-------------------------------------------
```

Comment:
I would move this sub-section on top, to have the same order as in “specification”